src/apps/api/views/competitions.py

```diff
@@ -212,45 +216,48 @@ def email_all_participants(self, request, pk):
-            for score in submission.scores.all():
-                leaderboard_data[leaderboard_titles[leaderID]][submission.owner.username].update({columns_titles[leaderID][f'col:{score.column_id}-task:{submission.task_id}']: float(score.score)})
+    @action(detail=True, methods=('GET',))
+    def leaderboard_submissions(self, request, pk):
+        comp = self.get_object()
+        phases = comp.phases.all()
+        return Response(self.get_serializer(phases, many=True).data)
+
+    def collect_leaderboard_data(self, competition, phase_pk=None):
+        if phase_pk:
+            phase = get_object_or_404(competition.phases.all(), id=phase_pk)
+            submission_query = [self.get_serializer(phase).data]
+            phase_id = phase.id
+        else:
+            phases = competition.phases.all()
+            submission_query = self.get_serializer(phases, many=True).data
+            phase_id = phases.first().id
+
+        leaderboard = Leaderboard.objects.prefetch_related('columns').get(phases=phase_id)
+        leaderboard_titles = {phase['id']: f'{leaderboard.title} - {phase["name"]}({phase["id"]})' for phase in submission_query}
+        leaderboard_data = {title: {} for title in leaderboard_titles.values()}
+
+        for phase in submission_query:
+            generated_columns = OrderedDict()
+            for task in phase['tasks']:
+                for col in leaderboard.columns.all():
+                    generated_columns.update({f'{col.key}-{task["id"]}': f'{task["name"]}({task["id"]})-{col.title}'})
+            for submission in phase['submissions']:
+                if submission["owner"] not in leaderboard_data[leaderboard_titles[phase['id']]].keys():
+                    leaderboard_data[leaderboard_titles[phase['id']]].update({submission["owner"]: OrderedDict()})
+                    for col_title in generated_columns.values():
+                        leaderboard_data[leaderboard_titles[phase['id']]][submission["owner"]].update({col_title: ""})
+                for score in submission['scores']:
+                    score_column = generated_columns[f'{score["column_key"]}-{submission["task"]}']
+                    leaderboard_data[leaderboard_titles[phase['id']]][submission["owner"]].update({score_column: score['score']})
         return leaderboard_data
 
     @action(detail=True, methods=['GET'], renderer_classes=[JSONRenderer, CSVRenderer, ZipRenderer])
     def results(self, request, pk, format=None):
         competition = self.get_object()
         if not competition.user_has_admin_permission(request.user):
             raise PermissionDenied("You are not a competition admin or superuser")
-        data = self.collect_leaderboard_data(competition)
-        selected_data = {}
-        selected_id = request.GET.get('id')
-        selected_leaderboard = request.GET.get('title')
+        selected_phase = request.GET.get('phase')
```

Comment:
Removed the option to filter leaderboard by title, and then added the option to get the leaderboard for a selected phase.